apps/openmw/mwrender/animation.cpp

```diff
@@ -460,6 +460,46 @@ namespace
 
 namespace MWRender
 {
+    class TransparencyUpdater : public SceneUtil::StateSetUpdater
+    {
+    public:
+        TransparencyUpdater(const float alpha)
+            : mAlpha(alpha)
+        {
+        }
+
+        void setAlpha(const float alpha)
+        {
+            mAlpha = alpha;
+        }
+
+    protected:
+        virtual void setDefaults(osg::StateSet* stateset)
+        {
+            osg::Material* material = static_cast<osg::Material*>(stateset->getAttribute(osg::StateAttribute::MATERIAL));
+
+            osg::BlendFunc* blendfunc (new osg::BlendFunc);
+            stateset->setAttributeAndModes(blendfunc, osg::StateAttribute::ON|osg::StateAttribute::OVERRIDE);
+
+            // FIXME: overriding diffuse/ambient/emissive colors
+            material = new osg::Material;
+            material->setColorMode(osg::Material::OFF);
+            material->setDiffuse(osg::Material::FRONT_AND_BACK, osg::Vec4f(1,1,1,mAlpha));
+            material->setAmbient(osg::Material::FRONT_AND_BACK, osg::Vec4f(1,1,1,1));
+            stateset->setAttributeAndModes(material, osg::StateAttribute::ON|osg::StateAttribute::OVERRIDE);
+            stateset->addUniform(new osg::Uniform("colorMode", 0), osg::StateAttribute::ON|osg::StateAttribute::OVERRIDE);
+        }
+
+        virtual void apply(osg::StateSet* stateset, osg::NodeVisitor* /*nv*/)
+        {
+            osg::Material* material = static_cast<osg::Material*>(stateset->getAttribute(osg::StateAttribute::MATERIAL));
+            material->setAlpha(osg::Material::FRONT_AND_BACK, mAlpha);
+        }
+
+    private:
+        float mAlpha;
```

Comment:
This isn't thread-safe, you should be using `std::atomic<float>` if you're updating and using the float from different threads

Reply:
The whole animation object is not supposed to be used in different threads, also we do not use atomic in other stateset updaters.

Reply:
If it's not used in different threads, then that's fine.